src/pointings_search/fits_utils.py

```diff
@@ -0,0 +1,179 @@
+
+    # Try to get the RA, dec of the center of the image.
+    sky_pos = wcs.pixel_to_world(int(num_r / 2), int(num_c / 2))
+
+    # Check that we produced a valid SkyCoord result.
+    if type(sky_pos) is list:
+        return None
+
+    # Compute an estimated FOV from the corners.
+    max_dist = np.max(
+        [
+            sky_pos.separation(wcs.pixel_to_world(0, 0)).degree,
+            sky_pos.separation(wcs.pixel_to_world(num_r, 0)).degree,
+            sky_pos.separation(wcs.pixel_to_world(0, num_c)).degree,
+            sky_pos.separation(wcs.pixel_to_world(num_r, num_c)).degree,
+        ]
+    )
+    return (sky_pos.ra.deg, sky_pos.dec.deg, max_dist)
+
+
+def obstime_from_hdu(hdu, default=-1.0):
+    """Extract an observation time from the FITS file using common
+    header keys: MJD, DATE-AVE.
+
+    Parameters
+    ----------
+    hdu :
+        The HDU to read.
+
+    Returns
+    -------
+    obstime : `float`
+        The obstime. This is set to default if no matching header
+        is found.
+    """
+    obstime = default
+    if "MJD" in hdu.header:
+        obstime = hdu.header["MJD"]
+    elif "DATE-AVE" in hdu.header:
+        obstime = hdu.header["DATE-AVE"]
+    return obstime
+
+
+def extend_pointing_dict_from_fits_file(filename, data_dict, extension=-1):
+    """Append to a pointings dictionary from a single FITS file. Checks either
+    a specific extension or all extensions.
+
+    Parameters
+    ----------
+    filename : `str`
+        The location of the file.
+    data_dict : `dict`
+        The pointing dictionary to extend. Contains ra, dec, obstime, filename,
+        layer, and fov. Modified in place.
+    extension : `int`
+        The extension in which to read the WCS and obstime. If no extension is
+        given it will try to read a WCS from all layers.
+
+    Returns
+    -------
+    num_added : `int`
+        The number of entries added.
+    """
+    num_added = 0
+    with io.fits.open(filename) as hdu_list:
+        # Check for a common (layer 0) obstime for all images.
+        base_mjd = obstime_from_hdu(hdu_list[0], -1.0)
+
+        # If we only want a single layer otherwise iterate through all layers.
+        if extension != -1:
+            if extension >= len(hdu_list):
+                return
+            layers_to_check = [extension]
+        else:
+            layers_to_check = [i for i in range(len(hdu_list))]
+
+        for i in layers_to_check:
+            # Read the WCS and skip this layer if there is an error.
+            coords = pointing_from_hdu(hdu_list[i])
+            if coords is None:
+                continue
+
+            # Check for an extension specific timestamp.
+            obstime = obstime_from_hdu(hdu_list[i], base_mjd)
+            if obstime < 0.0:
+                continue
+
+            # Append the data.
+            data_dict["ra"].append(coords[0])
+            data_dict["dec"].append(coords[1])
+            data_dict["obstime"].append(obstime)
```

Comment:
this is more of a down-the-line thing to worry about as I don't think it makes much difference for us in the kbmod case, but a lot of fits headers will have multiple observation times as observations can last on the order of 10s of seconds to minutes, i.e. they'll have beginning of observation + the midpoint + the length of exposure. you can ask colin for more info on this

Reply:
Agreed. If we want to support general FITS files, this will need a lot more logic.